### DISCUSSION TITLE
Preheat configuration not recognized after update

### DISCUSSION THREAD

**Discussion by martin-hei** [KlipperScreen/KlipperScreen#1258] [Q&A] [answered]
Hi all,
yesterday, I updated KlipperScreen to the most recent version. After that, I noticed, that the preheat configuration in KlipperScreen.conf is not longer recognized, I only get the standard configuration. There is nothing about this in the logs. Rebooting did not help. Any ideas?

Thanks a lot!
Martin

My configuration looks like this:

```
[preheat PLA]
extruder: 200
heater_bed: 60

[preheat PLA+]
extruder: 220
heater_bed: 60

[preheat Bed]
extruder: 0
heater_bed: 60

[preheat Hot-End]
extruder: 200
heater_bed: 0

[preheat PETG]
extruder: 235
heater_bed: 60

[preheat cooldown]
gcode: M107
```

**Comment by alfrix** [comment]
the config works, please post your logfile, and where is your configuration file located?

**Reply by martin-hei** [reply]
Hi @alfrix ,
thanks for your help. It pointed me to the solution.

For some reason, in line 26 of my "config.py", there is this line of code:
`printer_data_config = os.path.join(home, "printer_data", "printer_1_data", "config") # Changes`

This is different from the code in the repository, which looks like this:
`printer_data_config = os.path.join(home, "printer_data", "config")`

This of course leads to the wrong configuration folder (~/printer_data/printer_1_data/config), which doesn't even exist.
Do you have any idea, what process would change the config.py? It even left a comment ;-)

I only have one instance of plain vanilla Klipper running on my FLSUN SpeederPad.

Thanks!
Martin

**Reply by alfrix** [reply, marked as answer]
the seems to come from @Guilouz fork, the os.path.join function concatenates a path, he would need to replace the "printer_data" to "printer_1_data",  to me the best place to add all of those extra dirs is if we wants them are in 
https://github.com/KlipperScreen/KlipperScreen/blob/71ffa005916d0485db17d6cd9a67a6172831e51c/ks_includes/config.py#L410

i don't support this, since klipperscreen only uses 1 config file, it wouldn't make sense to have the config repeated in multiple printer_n_data folders, but to each their own, the user can symbolic links if they prefer to have the config in all the folders

related info:
where does klipperscreen search config and the order
https://klipperscreen.github.io/KlipperScreen/Configuration/#klipperscreen-behaviour-towards-configuration

**Reply by Guilouz** [reply]
Thanks for the tips.

I have added printer_1_data because in case of multi-printers there is no printer_data folder and starts with printer_1_data, printer_2_data etc... To put klipperscreen.conf file in printer_1_data folder, it's more easy for users to have this file can be edited in their config folder on web UI.

So can I use ?

printer_1_data_config = os.path.join(home, "printer_1_data", "config")
directories = [printer_data_config, printer_1_data_config, xdg_config, klipperscreendir]

**Reply by alfrix** [reply]
> So can I use ?
> 
> printer_1_data_config = os.path.join(home, "printer_1_data", "config")
> directories = [printer_data_config, printer_1_data_config, xdg_config, klipperscreendir]

yes, that should work
alternatively you can make a for that checks this case *first* for example:

```
# Check for multi printer setups first
for i in range(1, 10):
            path = self.check_path_exists(os.path.join(home, f"printer_{i}_data", "config"), self.configfile_name)
            if path:
                return path
```

**Reply by alfrix** [reply]
but if you only want to check printer_1_data i would prefer the way you did it
